core/controllers/acl_decorators.py

```diff
@@ -2318,11 +2386,22 @@ def test_can_access(self, topic_name, **kwargs):
         Raises:
             PageNotFoundException: The given page cannot be found.
         """
-        topic_name = python_utils.url_unquote_plus(topic_name)
-        topic = topic_fetchers.get_topic_by_name(topic_name)
+        topic = topic_fetchers.get_topic_by_abbreviated_name(
+            abbreviated_topic_name)
+        classroom_name = kwargs.pop('classroom_name', None)
 
         if topic is None:
-            raise self.PageNotFoundException
+            redirect_permanently(
```

Comment:
Ditto as above, please do not redirect any error cases permanently. Here and elsewhere.